yt_dlp/extractor/ntvcojp.py

```diff
@@ -1,55 +1,83 @@
-from .common import InfoExtractor
+from .streaks import StreaksBaseIE
 from ..utils import (
-    ExtractorError,
-    smuggle_url,
-    traverse_obj,
+    int_or_none,
+    parse_iso8601,
+    str_or_none,
+    url_or_none,
 )
+from ..utils.traversal import require, traverse_obj
 
 
-class NTVCoJpCUIE(InfoExtractor):
-    IE_NAME = 'cu.ntv.co.jp'
-    IE_DESC = 'Nippon Television Network'
-    _VALID_URL = r'https?://cu\.ntv\.co\.jp/(?!program)(?P<id>[^/?&#]+)'
-    _TEST = {
-        'url': 'https://cu.ntv.co.jp/televiva-chill-gohan_181031/',
+class NTVJpCuIE(StreaksBaseIE):
+    IE_NAME = 'ntvjp:cu'
+    IE_DESC = '日テレ無料TADA!'
+
+    _VALID_URL = r'https?://cu\.ntv\.co\.jp/(?!program-list)(?P<id>[\w-]+)/?$'
```

Comment:
why the `/?$` ??
```suggestion
    _VALID_URL = r'https?://cu\.ntv\.co\.jp/(?!program-list)(?P<id>[\w-]+)'
```

Reply:
to avoid matching program URLs
```suggestion
    _VALID_URL = r'https?://cu\.ntv\.co\.jp/(?!program-list)(?P<id>[\w-]+)/?(?:[?#]|$)'
```